ProductService.js

```diff
@@ -0,0 +1,110 @@
+class Product {
+    constructor(name, destription, price, tags, images) {
+        this.name = name;
+        this.destription = destription;
+        this.price = price;
+        this.tags = tags;
+        this.images = images;
+        this.favoriteCount = 0;
+    }
+    favorite() {
+        this.favoriteCount++;
+    }
+}
+
+class ElectronicProduct extends Product {
+    (name, destription, price, tags, images, manufacturer) {
+        super (name, destription, price, tags, images);
+        this.manufacturer = manufacturer;
+    }
+}
+
+const BASE_URL = 'https://panda-market-api-crud.vercel.app';
+
+async function getProductList (page = 1, pageSize = 10, keyword = '') {
+    try {
+        const response = await fetch (`${BASE_URL}/products?page=${page}&pageSize=
+            ${pageSize}&keyword=${keyword}`);
+        if (!response.ok) {
+            throw new Error (`HTTP error! status: ${response.status}`);
+        }
+        const data = await response.json();
+        const products = data.map (item => {
+            if (item.tag.includes ('전자제품')) {
+                return new ElectronicProduct(item.name, item.destription, item.price, item.tags,
+                     item.images, item.manufacturer);
+            } else {
+                return new Product (item.name, item.destription, item.price, item.tags, item.images)
+            }
+        });
+        return products;
+    } catch (error) {
+        console.error('Error fetching product list:', error);
+    }
+}
+
+async function getProduct(productId) {
+    
+    try {
+        const response = await fetch(`${BASE_URL}/products/${productId}`);
+        if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        return await response.json();
+    } catch (error) {
+        console.error(`Error fetching product ${productId}:`, error);
+    }
```

Comment:
error를 log로 남기고, 다시 throw 하게 되면, 이 함수를 호출한 곳에서 또 핸들링이 가능합니다. 이렇게 throw 하지 않으면 로그로만 해당 에러를 알 수 있어서 디버깅에 어려움을 겪을 수 있어요.


```suggestion
    } catch (error) {
        console.error(`Error fetching product ${productId}:`, error);

        throw error
    }
```